### ISSUE TITLE
Suggestion on 'Submodels' explanation and question on 632 BS and inference

### ISSUE THREAD

**Issue by sagesteppe** [simonpcouch/slc-rug-25#1]
Hey Simon Great talk today!

The suggestions you made to incorporate submodels into a hyperparameter tuning workflow have worked out great so far! Or at least I think, I don't too much bench-marking here...  

Anyways, it took me a couple readings of chapter 5 of your text to understand the submodel trick; I didn't quite grasp it during the talk (it could have been that it was because I was between sessions). Actually, it took me reading the note in Section 5.1.2 to *pick up what you're putting down*.  And well now it's painfully obvious! 

I'm going to just try and get the ball rolling here and wonder if some notes to the effect of the following may help other readers. 

### For section 5.1.2. 
*(Figure refers to the last plot in 5.1.1)*

"In this example (Figure XX), for each of the four levels of learn rate which we will be tuned, we will fit only the maximum number of trees (2000). Because the engine will pass through each level of trees required (1, 667, 1333) on the way to 2000, all combinations of the trees hyperparameter will be visited for each learning rate in a single pass. Thus by starting with the max trees, rather than the minimum number of trees and working 'upwards', the number of total models predicted becomes 1/4 those which are present in our grid. "

Basically, it's not very explicit that you simply fit the max tree model first. I'm sure there are other ways to make it more clear as well, but because the paragraph starts with the assumption fewest trees first, and then you flip that on it's head it could be good to have some notation on that. 

### 5.1.2. note

However, this does not result in a 4x speed-up contribution from this trick because the length of time required to fit a BRT scales linearly with the number of trees. Rather the speed up is theoretically only roughly 2x (1 + 667 + 1333 + 2000 = 4000 INSTEAD 4000/2000) blah blah overhead. (right?)

----

Finally this may be overkill, but I think a fun way of visualizing the trick could be an arc diagram.  Hopefully these help! Otherwise book is looking good so far. I suppose the next question people may have is how does the submodelling trick scale while tuning more than two hyperparameters? You did use XGboost as the example - so you kind of have that coming! 

----

My next question, not super related to your talk, is about inference from predictive models using .632 bootstrap. I have read a couple of resources on 632 bootstrap, but I am not seeing an explicit connection to making inference, would you happen to have any resource on this? Or did I just hear something wrong during Q&A?

**Comment by simonpcouch** [comment]
Ah, this is great feedback! Thanks for the suggestion—I'll incorporate those changes into the chapter next time I'm focused on it.

I don't remember exactly the context in which the 632/632+ came up in the discussion, but the extent to which we've considered incorporating it in tidymodels is discussed [here](https://aml4td.org/chapters/resampling.html#correcting-for-bias)—debiasing performance estimates while resampling. Apologies if my description of that was confusing the Q&A!

**Comment by sagesteppe** [comment]
Thanks for sharing that bookdown and the relevant discussion! I've only been able to glance through it, but I am quite intrigued with what they suggest it can do. Any idea what the timeline is on implementing this?
